runtime/hydradx/src/migrations/reset_max_nonces.rs

```diff
@@ -0,0 +1,1995 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Migration to reset nonces from u32::MAX to correct values for affected accounts
+//!
+//! This migration fixes a bug where 1,802 accounts ended up with nonce = u32::MAX (4294967295).
+//! The affected accounts were identified on 2025-12-09 (1,802 accounts) via chain scans.
+//! Each account's nonce is set to either:
+//! - Their EVM transaction count (if they have an EVM address in the transaction log)
+//! - 0 (if they don't have an EVM address or no transactions found)
+
+use frame_support::{
+	storage_alias,
+	traits::{Get, OnRuntimeUpgrade},
+	weights::Weight,
+};
+
+#[cfg(feature = "try-runtime")]
+use codec::{Decode, Encode};
+
+pub mod v1 {
+	use super::*;
+	use crate::AccountId;
+	use sp_core::crypto::Ss58Codec;
+
+	/// Storage to track if this migration has been executed
+	#[storage_alias]
+	pub type ResetMaxNoncesMigrationExecuted<T: frame_system::Config> = StorageValue<frame_system::Pallet<T>, bool>;
+
+	pub struct ResetMaxNonces<T>(sp_std::marker::PhantomData<T>);
+
+	impl<T: frame_system::Config<AccountId = AccountId, Nonce = u32>> OnRuntimeUpgrade for ResetMaxNonces<T> {
+		fn on_runtime_upgrade() -> Weight {
+			// Check if migration has already been executed
+			if ResetMaxNoncesMigrationExecuted::<T>::get().unwrap_or(false) {
+				log::info!(
+					target: "runtime::reset_max_nonces",
+					"Migration already executed, skipping..."
+				);
+				return T::DbWeight::get().reads(1);
+			}
+
+			log::info!(
+				target: "runtime::reset_max_nonces",
+				"Starting migration to reset {} accounts with max nonce",
+				AFFECTED_ACCOUNTS.len()
+			);
+
+			let mut reads = 1u64; // Already read the migration flag
+			let mut writes = 1u64; // Will write the migration flag
+			let mut reset_count = 0u64;
+			let mut skipped_count = 0u64;
+
+			for (account_str, new_nonce) in AFFECTED_ACCOUNTS.iter() {
+				// Decode the SS58 address to AccountId
+				let account_id = match AccountId::from_ss58check(account_str) {
+					Ok(acc) => acc,
+					Err(_) => {
+						log::warn!(
+							target: "runtime::reset_max_nonces",
+							"Failed to decode address: {}",
+							account_str
+						);
+						skipped_count += 1;
+						continue;
+					}
+				};
+
+				// Mutate the account info atomically
+				reads += 1;
+				let was_reset = frame_system::Account::<T>::mutate(&account_id, |account_info| {
+					// Check if nonce is u32::MAX
+					if account_info.nonce == u32::MAX {
+						// Set nonce to the specified value
+						account_info.nonce = *new_nonce;
+						true
+					} else {
+						false
+					}
+				});
+
+				if was_reset {
+					writes += 1;
+					reset_count += 1;
+
+					if reset_count % 100 == 0 {
+						log::info!(
+							target: "runtime::reset_max_nonces",
+							"Reset {} nonces so far...",
+							reset_count
+						);
+					}
+				} else {
+					skipped_count += 1;
+				}
+			}
```

Comment:
this is quite hard to read. 

i am not really sure why we need to extra storage to check if this was executed or not. 

i dont think it is necessary. 

suggestion to rewrite the code. 

```suggestion
			for (account_str, new_nonce) in AFFECTED_ACCOUNTS {
    let Ok(account_id) = AccountId::from_ss58check(account_str) else {
        log::warn!(
            target: "runtime::reset_max_nonces",
            "Failed to decode address: {account_str}"
        );
        skipped_count += 1;
        continue;
    };

    reads += 1;
    
    let was_reset = frame_system::Account::<T>::mutate(&account_id, |info| {
        (info.nonce == u32::MAX).then(|| info.nonce = *new_nonce).is_some()
    });

    match was_reset {
        true => {
            writes += 1;
            reset_count += 1;
            if reset_count % 100 == 0 {
                log::info!(
                    target: "runtime::reset_max_nonces",
                    "Reset {reset_count} nonces so far..."
                );
            }
        }
        false => skipped_count += 1,
    }
}
```